responses/responses.go

```diff
@@ -0,0 +1,135 @@
+
+// MovedPermanently HTTP 301 response.
+func MovedPermanently() huma.Response {
+	return response(http.StatusMovedPermanently)
+}
+
+// Found HTTP 302 response.
+func Found() huma.Response {
+	return response(http.StatusFound)
+}
+
+// NotModified HTTP 304 response.
+func NotModified() huma.Response {
+	return response(http.StatusNotModified)
+}
+
+// TemporaryRedirect HTTP 307 response.
+func TemporaryRedirect() huma.Response {
+	return response(http.StatusTemporaryRedirect)
+}
+
+// PermanentRedirect HTTP 308 response.
+func PermanentRedirect() huma.Response {
+	return response(http.StatusPermanentRedirect)
+}
+
+// BadRequest HTTP 400 response with a structured error body (e.g. JSON).
+func BadRequest() huma.Response {
+	return errorResponse(http.StatusBadRequest)
+}
+
+// Unauthorized HTTP 401 response with a structured error body (e.g. JSON).
+func Unauthorized() huma.Response {
+	return errorResponse(http.StatusUnauthorized)
+}
+
+// Forbidden HTTP 403 response with a structured error body (e.g. JSON).
+func Forbidden() huma.Response {
+	return errorResponse(http.StatusForbidden)
+}
+
+// NotFound HTTP 404 response with a structured error body (e.g. JSON).
+func NotFound() huma.Response {
+	return errorResponse(http.StatusNotFound)
+}
+
+// RequestTimeout HTTP 408 response with a structured error body (e.g. JSON).
+func RequestTimeout() huma.Response {
+	return errorResponse(http.StatusRequestTimeout)
+}
+
+// Conflict HTTP 409 response with a structured error body (e.g. JSON).
+func Conflict() huma.Response {
+	return errorResponse(http.StatusConflict)
+}
+
+// PreconditionFailed HTTP 412 response with a structured error body (e.g. JSON).
+func PreconditionFailed() huma.Response {
+	return errorResponse(http.StatusPreconditionFailed)
+}
+
+// RequestEntityTooLarge HTTP 413 response with a structured error body (e.g. JSON).
+func RequestEntityTooLarge() huma.Response {
+	return errorResponse(http.StatusRequestEntityTooLarge)
+}
+
+// PreconditionRequired HTTP 428 response with a structured error body (e.g. JSON).
+func PreconditionRequired() huma.Response {
+	return errorResponse(http.StatusPreconditionRequired)
+}
+
+// InternalServerError HTTP 500 response with a structured error body (e.g. JSON).
+func InternalServerError() huma.Response {
+	return errorResponse(http.StatusInternalServerError)
+}
+
+// BadGateway HTTP 502 response with a structured error body (e.g. JSON).
+func BadGateway() huma.Response {
+	return errorResponse(http.StatusBadGateway)
+}
+
+// ServiceUnavailable HTTP 503 response with a structured error body (e.g. JSON).
+func ServiceUnavailable() huma.Response {
+	return errorResponse(http.StatusServiceUnavailable)
+}
+
+// GatewayTimeout HTTP 504 response with a structured error body (e.g. JSON).
+func GatewayTimeout() huma.Response {
+	return errorResponse(http.StatusGatewayTimeout)
+}
+
+// String HTTP response with the given status code and `text/plain` content
+// type.
+func String(status int) huma.Response {
```

Comment:
Is there a way to return a JSON-formatted error response? This is something we've had requested.

Reply:
Yep, any time you use `ctx.AddError` or `ctx.WriteError` you will get a nicely formatted [RFC 7807](https://tools.ietf.org/html/rfc7807) JSON error returned. If you use the built-in JSON Schema-based validation you even get enhanced errors with the specific JSON Path location and value along with why the validation failed at that location, e.g. `body.foo.bar` and `Unable to parse boolean: abc`.